src/Tree/RepositoryInterface.php

```diff
@@ -25,39 +27,39 @@ interface RepositoryInterface extends RepositoryUtilsInterface
      * @param string $sortByField
      * @param string $direction
      *
-     * @return array
+     * @return T[]
      */
     public function getRootNodes($sortByField = null, $direction = 'asc');
 
     /**
      * Returns an array of nodes optimized for building a tree.
      *
-     * @param object $node        Root node
+     * @param T $node        Root node
      * @param bool   $direct      Flag indicating whether only direct children should be retrieved
      * @param array  $options     Options, see {@see RepositoryUtilsInterface::buildTree()} for supported keys
      * @param bool   $includeNode Flag indicating whether the given node should be included in the results
      *
-     * @return array
+     * @return T[]
      */
     public function getNodesHierarchy($node = null, $direct = false, array $options = [], $includeNode = false);
 
     /**
      * Get the list of children for the given node.
      *
-     * @param object|null          $node        If null, all tree nodes will be taken
+     * @param T|null          $node        If null, all tree nodes will be taken
      * @param bool                 $direct      True to take only direct children
      * @param string|string[]|null $sortByField Field name or array of fields names to sort by
      * @param string|string[]      $direction   Sort order ('ASC'|'DESC'). If $sortByField is an array, this may also be an array with matching number of elements
      * @param bool                 $includeNode Include the root node in results?
      *
-     * @return array|null List of children or null on failure
+     * @return T[]|null List of children or null on failure
```

Comment:
Maybe this could be `iterable<array-key, T>` to allow any iterable to be returned?

Reply:
this would widen the return type, which is a BC break